mypy/semanal.py

```diff
@@ -2406,23 +2407,20 @@ def lookup_qualified(self, name: str, ctx: Context) -> SymbolTableNode:
 
     def builtin_type(self, fully_qualified_name: str) -> Instance:
         node = self.lookup_fully_qualified(fully_qualified_name)
-        info = cast(TypeInfo, node.node)
-        return Instance(info, [])
+        assert isinstance(node.node, TypeInfo)
+        return Instance(node.node, [])
 
     def lookup_fully_qualified(self, name: str) -> SymbolTableNode:
```

Comment:
Actually, this PR is pretty big already (and is not without risk of breakage -- changing casts to asserts potentially changes runtime behavior), so I'd prefer to keep it as focused as possible.  If you'd like to make this refactor, could you submit it as a separate small PR?